### ISSUE TITLE
Error while running your last action (Invalid request method).

### ISSUE THREAD

**Issue by starbock** [githubjeka/tracker-issues#41]
**Tracker-Issues:**  Latest version: 0.2.3 (installed from marketplace)
**HumHub  Community Edition:**  Version: 1.7.1
**PHP:**  7.4

**_To Reproduce:_**

- Click "Tracker Issues" from top navbar
- Click "+ New Issue" button
- Either enter Space or not in the New Issue Add Space input box (doesn't matter either way)
- Click "Add" button

**_Result_**

- At the bottom of the page I see the "Error while running your last action (Invalid request method)." error message.
- The bottom can be opened to reveal this trace information:
```
{
    "url": "/index.php?r=tracker-issues%2Fissue%2Fcreate&uguid=954284d0-afb3-42a6-9627-c033872c2df2",
    "status": 405,
    "response": {
        "error": true,
        "message": "Module is not enabled on this content container!"
    },
    "textStatus": "error",
    "xhr": {
        "readyState": 4,
        "responseText": "{\"error\":true,\"message\":\"Module is not enabled on this content container!\"}",
        "responseJSON": {
            "error": true,
            "message": "Module is not enabled on this content container!"
        },
        "status": 405,
        "statusText": "error"
    },
    "dataType": "json",
    "error": true,
    "message": "Module is not enabled on this content container!",
    "errorThrown": "",
    "validationError": false
}
```
There is no additional logging information in the HumHub log.

**Comment by githubjeka** [comment]
I thought the latest version automatically hit the market. But there everything is configured manually. 

>  Review notice: The new version must first be checked and approved by our staff. This can take up to 3 working days.

**Comment by ArchBlood** [comment]
> I thought the latest version automatically hit the market. But there everything is configured manually. 
> 
> >  Review notice: The new version must first be checked and approved by our staff. This can take up to 3 working days.

Apparently they've been working on changing the process on releases as well as published modules :thinking:

**Comment by githubjeka** [comment]
Fixed by v0.3.0 
The module is available in the market.

**Comment by starbock** [comment]
Still seeing after installing 0.3.0:

```
{
    "url": "/index.php?r=tracker-issues%2Fissue%2Fcreate&uguid=954284d0-afb3-42a6-9627-c033872c2df2",
    "status": 405,
    "response": {
        "error": true,
        "message": "Module is not enabled on this content container!"
    },
    "textStatus": "error",
    "xhr": {
        "readyState": 4,
        "responseText": "{\"error\":true,\"message\":\"Module is not enabled on this content container!\"}",
        "responseJSON": {
            "error": true,
            "message": "Module is not enabled on this content container!"
        },
        "status": 405,
        "statusText": "error"
    },
    "dataType": "json",
    "error": true,
    "message": "Module is not enabled on this content container!",
    "errorThrown": "",
    "validationError": false
}
```

**_Note:_**  The upgrade came in as a different module in the marketplace, meaning that the HumHub installation didn't recognize it as an upgrade.  I removed v0.2.3 and then installed v0.3.0.  This worked fine for me, as this is a fresh install, but it wouldn't have worked if I'd had any content.

**Comment by githubjeka** [comment]
When a module is deleted, all data of module in the database is also deleted. Try to clear directory `/protected/runtime/` to next time.

**Comment by starbock** [comment]
> When a module is deleted, all data of module in the database is also deleted. Try to clear directory `/protected/runtime/` to next time.

Thanks for your help with this.  I don't have a lot of runtime with HumHub, so I ask before making a mistake.

- I disabled and uninstalled the module
- I flushed the cache via `Administration -> Settings -> Advanced -> Caching - > Save & Flush`
- I re-installed the module
- Same behavior

When you say "clear directory /protected/runtime" do you mean to literally `rm -rf protected/runtime/*`, or do you mean something else?

Thanks!

**Comment by githubjeka** [comment]
Yes, `rm -rf protected/runtime/*`  fixes a bunch of problems at once usually.

**Comment by starbock** [comment]
I did it.  Unfortunately, same results:

```
{
    "url": "/index.php?r=tracker-issues%2Fissue%2Fcreate&uguid=954284d0-afb3-42a6-9627-c033872c2df2",
    "status": 405,
    "response": {
        "error": true,
        "message": "Module is not enabled on this content container!"
    },
    "textStatus": "error",
    "xhr": {
        "readyState": 4,
        "responseText": "{\"error\":true,\"message\":\"Module is not enabled on this content container!\"}",
        "responseJSON": {
            "error": true,
            "message": "Module is not enabled on this content container!"
        },
        "status": 405,
        "statusText": "error"
    },
    "dataType": "json",
    "error": true,
    "message": "Module is not enabled on this content container!",
    "errorThrown": "",
    "validationError": false
}
```
Is there some other information that I can provide?  Thanks for your help!

**Comment by githubjeka** [comment]
What are the default settings?

![image](https://user-images.githubusercontent.com/874234/101653615-8f3f0a00-3a50-11eb-9ebf-6389eba9b2c5.png)

**Comment by starbock** [comment]
Yes, that was the issue, which raises the following for me:

- I don't know if this is under your control or if HumHub handles this itself.  As a suggestion, the error handler / error message could be improved to allow for better self-troubleshooting.  In hindsight now I get it, but as a newcomer to HumHub it's not immediately obvious ... and it would be very confusing to end users.

> **Use Case** - We want to enable / disable a given user from being able to submit tickets.  In this scenario that we disable, the user would get a user-friendly message on the lines of, "Your account is not authorized to submit tickets." or something similar.  That would have helped me in my self-troubleshooting steps.

- This raises a question in my mind.  What is the meaning of deactivating in a Space?  Is this what you intended?

> **Use Case** - We want to have Space-specific tickets, where the group working in a Space are able to create tickets and have a view that shows them only their tickets, thus enabling Space-level support that is not distracted by other Space's issues.

- FYI, although it appears to be a HumHub issue, not module-specific ...  If I set it to Activated, when the user tries to disable there is a warning that all the profile's data will be deleted, but there is no button to agree or cancel.  And, closing that dialog leaves the modules display ( `<Space> | Modules` ) with a cycling progress `Disable` button for the module ... I had to refresh the page to clear this condition, and it's reproducible.

Thank you for helping me sort this piece out.  At least you (and others) got the benefit of getting the marketplace updated from this exercise.  I appreciate your time and contributions.
